### PR TITLE
Add enable_abort_on_exception

### PR DESCRIPTION
**Pull request by elauksap**
As the title says.

### CODE REVIEW

File: include/deal.II/base/exceptions.h [hunk 1]
```diff
@@ -1286,11 +1286,23 @@ namespace deal_II_exceptions
...
+   * @see enable_abort_on_exception
    * @see Exceptions
    */
   void
   disable_abort_on_exception();
 
+  /**
+   * Calling this function switches on the use of <tt>std::abort()</tt> when
+   * an exception is created using the Assert() macro, instead of throwing it.
+   * This restores the standard behavior.
```
Comment (peterrum) [inline include/deal.II/base/exceptions.h:1298]: What is the standard behavior?

**Review by peterrum — COMMENTED** [review]
Any opinions here? I am not an expert on exceptions...

File: include/deal.II/base/exceptions.h (hunk 1 above)
Comment (elauksap) [reply include/deal.II/base/exceptions.h:1298]: The default behavior is to abort when an exception is thrown.

Before this PR, this behavior could be disabled via `disable_abort_on_exception` but there was no analogous way to reset it (except by explicitly setting the public attribute `internals::allow_abort_on_exception = true;`, which is a bit messy in my opinion).

**Review by bangerth — APPROVED** [review]

**Comment by bangerth** [comment]
This makes sense to me. The reverse function already exists.

/rebuild

**Pull request merged**